docs/src/upgrade/release-3.7.x.asciidoc

```diff
@@ -56,6 +56,23 @@ drc.open().catch(err => {
 })
 ----
 
+==== Creation of New `gremlin-util` Module
+
+`gremlin-driver` has been refactored and several classes have been extracted to a new `gremlin-util` module. Any classes
+which are utilized by both `gremlin-driver` and `gremlin-server` have been extracted to `gremlin-util`. This includes
+the entire `tinkerpop.gremlin.driver.ser` and `tinkerpop.gremlin.driver.message` packages as well as
+`tinkerpop.gremlin.driver.MessageSerializer` and `tinkerpop.gremlin.driver.Tokens`. This change is non-breaking as the
+package names remain unchanged and `gremlin-util` remains included in `gremlin-driver` as a dependency. As long as
+maven is being used as a build system, there is no change to the use in any of these classes.
+
+See: link:https://issues.apache.org/jira/browse/TINKERPOP-2819[TINKERPOP-2819]
+
+==== Reworked Gremlin Socket Server
```

Comment:
i think this section is of more interest to providers than it is to users. could you please move it down there? it looks like the section headers for "providers" is sorta messed up too. could you please look at the other versions and mimic that structure here?